flyteadmin/pkg/common/executions.go

```diff
@@ -1,20 +1,33 @@
 package common
 
 import (
+	"context"
 	"fmt"
 
+	"github.com/wolfeidau/humanhash"
 	"k8s.io/apimachinery/pkg/util/rand"
 
 	"github.com/flyteorg/flyte/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flyte/flytestdlib/logger"
 )
 
 const ExecutionIDLength = 20
+const ExecutionIDLengthLimit = 63
 const ExecutionStringFormat = "a%s"
 
 /* #nosec */
-func GetExecutionName(seed int64) string {
+func GetExecutionName(seed int64, enableHumanHash bool) (string, error) {
 	rand.Seed(seed)
-	return fmt.Sprintf(ExecutionStringFormat, rand.String(ExecutionIDLength-1))
+	if enableHumanHash {
+		hashKey := []byte(rand.String(ExecutionIDLength))
+		result, err := humanhash.Humanize(hashKey, 3)
```

Comment:
IIUC, by setting this to 3 we get about 16M unique ids to choose from. The reason is that humanhash maintains a list of [unique 256 words](https://github.com/wolfeidau/humanhash/blob/master/human_hash.go#L28-L64), so 256^3 = 16.8M. 

We should have a conversation about whether this is enough for most big deployments. My gut feeling is that it's not sufficient. How about we extend this to 4? If we do this and end up choosing the largest word in that set 4 times we're still under the 63, which is the max char limit for an execution id. Out of curiosity, the largest word in that dataset is "massachusetts" at 13 characters, so that would be 55 characters (13*4 + 3 to account for the separators).